### ISSUE TITLE
DesiredCapabilities::__construct() must be of the type array, null given

### ISSUE THREAD

**Issue by poriia** [facebook/php-webdriver#603]
i have problem in first dusk test!
after run artisan dusk, show this error:
1) Tests\Browser\ExampleTest::testBasicExample
TypeError: Argument 1 passed to Facebook\WebDriver\Remote\DesiredCapabilities::__construct() must be of the type array, null given, called in ../vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php on line 127

dusk prepare is empty.
dusk driver:
return RemoteWebDriver::create('http://nayket-server.local', DesiredCapabilities::chrome());

APP_URL is nayket-server.local

nginx service is run... and i see home page in chrome browser.

PHP 7.2.9-1+ubuntu16.04

**Comment by shadyueh** [comment]
I'm having similar problem with codeception test and yii1.
The facebook/webdriver is 1.6.0.
```
[PHPUnit_Framework_Exception] Argument 1 passed to Facebook\WebDriver\Remote\DesiredCapabilities::__construct() must be of the type array, null given, called in /var/www/html/dsv/test-app/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php on line 127 and defined
```
Please, fix this.

```
$ composer show
behat/gherkin                      v4.5.1             Gherkin DSL parser for PHP 5.3
codeception/codeception            2.5.0              BDD-style testing framework
codeception/phpunit-wrapper        6.0.12             PHPUnit classes used by Codeception
codeception/stub                   2.0.4              Flexible Stub wrapper for PHPUnit's Mock Builder
codeception/yii-bridge             dev-master a5841d6 YiiBridge extension
doctrine/instantiator              1.0.5              A small, lightweight utility to instantiate objects in PHP without invoking their constructors
facebook/webdriver                 1.6.0              A PHP client for Selenium WebDriver
guzzlehttp/guzzle                  6.3.3              Guzzle is a PHP HTTP client library
guzzlehttp/promises                v1.3.1             Guzzle promises library
guzzlehttp/psr7                    1.4.2              PSR-7 message implementation that also provides common utility methods
myclabs/deep-copy                  1.7.0              Create deep copies (clones) of your objects
phpdocumentor/reflection-common    1.0.1              Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock  3.3.2              With this component, a library can provide support for annotations via DocBlocks or otherwise retrieve information that is embedded in a DocBlock.
phpdocumentor/type-resolver        0.4.0             
phpspec/prophecy                   1.8.0              Highly opinionated mocking framework for PHP 5.3+
phpunit/php-code-coverage          4.0.8              Library that provides collection, processing, and rendering functionality for PHP code coverage information.
phpunit/php-file-iterator          1.4.5              FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-text-template          1.2.1              Simple template engine.
phpunit/php-timer                  1.0.9              Utility class for timing
phpunit/php-token-stream           1.4.12             Wrapper around PHP's tokenizer extension.
phpunit/phpunit                    5.7.27             The PHP Unit Testing framework.
phpunit/phpunit-mock-objects       3.4.4              Mock Object library for PHPUnit
psr/http-message                   1.0.1              Common interface for HTTP messages
psr/log                            1.0.2              Common interface for logging libraries
sebastian/code-unit-reverse-lookup 1.0.1              Looks up which function or method a line of code belongs to
sebastian/comparator               1.2.4              Provides the functionality to compare PHP values for equality
sebastian/diff                     1.4.3              Diff implementation
sebastian/environment              2.0.0              Provides functionality to handle HHVM/PHP environments
sebastian/exporter                 2.0.0              Provides the functionality to export PHP variables for visualization
sebastian/global-state             1.1.1              Snapshotting of global state
sebastian/object-enumerator        2.0.1              Traverses array structures and object graphs to enumerate all referenced objects
sebastian/recursion-context        2.0.0              Provides functionality to recursively process PHP variables
sebastian/resource-operations      1.0.0              Provides a list of PHP built-in functions that operate on resources
sebastian/version                  2.0.1              Library that helps with managing the version number of Git-hosted PHP projects
symfony/browser-kit                v3.4.17            Symfony BrowserKit Component
symfony/console                    v3.4.17            Symfony Console Component
symfony/css-selector               v3.4.17            Symfony CssSelector Component
symfony/debug                      v3.4.17            Symfony Debug Component
symfony/dom-crawler                v3.4.17            Symfony DomCrawler Component
symfony/event-dispatcher           v3.4.17            Symfony EventDispatcher Component
symfony/finder                     v3.4.17            Symfony Finder Component
symfony/polyfill-ctype             v1.9.0             Symfony polyfill for ctype functions
symfony/polyfill-mbstring          v1.9.0             Symfony polyfill for the Mbstring extension
symfony/process                    v3.4.17            Symfony Process Component
symfony/yaml                       v3.4.17            Symfony Yaml Component
webmozart/assert                   1.3.0              Assertions to validate method input/output with nice error messages.
```

**Comment by jhae-de** [comment]
I have the same problem with facebook/webdriver version 1.7.1.

A simple `RemoteWebDriver::create()` results in:

`TypeError: Argument 1 passed to Facebook\WebDriver\Remote\DesiredCapabilities::__construct() must be of the type array, null given, called in /app/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php on line 145`

**Comment by OndraM** [comment]
This may be caused by missing implementation W3C protocol (see #469), and should be fixed in upcoming release 1.8.0.
